File: sdk/csharp/examples/Dns.cs
// DNS operations with the HyperMesh C# SDK.
//
// Run: dotnet run

using HyperMesh.Sdk;

using var client = new HyperMeshClient();

// List existing DNS records
var dns = await client.Dns.ListAsync();
Console.WriteLine($"DNS records: {dns.Count}");
foreach (var record in dns.Records)
{
    Console.WriteLine($"  {record.Name} -> {record.Address}");
}

// Register a new record
Console.WriteLine("\nRegistering example.hypermesh -> ::1");
var reg = await client.Dns.RegisterAsync("example.hypermesh", "::1");
Console.WriteLine($"Registered: {reg}");

// Resolve the record
try
{
    var resolved = await client.Dns.ResolveAsync("example.hypermesh");
    Console.WriteLine($"Resolved: {resolved.Name} -> {resolved.Address}");
}
catch (HyperMeshException ex)
{
    Console.WriteLine($"Resolve failed: {ex.Message}");
}

// List records after registration
var updated = await client.Dns.ListAsync();
Console.WriteLine($"\nDNS records after registration: {updated.Count}");
foreach (var record in updated.Records)
{
    Console.WriteLine($"  {record.Name} -> {record.Address}");
}

File: sdk/cpp/examples/dns.cpp
// DNS operations with the HyperMesh C++ SDK.
//
// Build: g++ -std=c++17 -I../include dns.cpp ../src/*.cpp ../src/api/*.cpp -o dns
// Run:   ./dns

#include <hypermesh/client.hpp>
#include <hypermesh/error.hpp>
#include <iostream>

int main() {
    hypermesh::HyperMeshClient client;

    // List existing DNS records
    auto dns = client.dns().list();
    std::cout << "DNS records: " << dns.count << "\n";
    for (const auto& record : dns.records) {
        std::cout << "  " << record.name << " -> " << record.address << "\n";
    }

    // Register a new record
    std::cout << "\nRegistering example.hypermesh -> ::1\n";
    auto reg = client.dns().register_record("example.hypermesh", "::1");
    std::cout << "Register result: " << reg.data.dump() << "\n";

    // Resolve the record
    try {
        auto resolved = client.dns().resolve("example.hypermesh");
        std::cout << "Resolved: " << resolved.name << " -> " << resolved.address << "\n";
    } catch (const hypermesh::HyperMeshError& e) {
        std::cout << "Resolve failed: " << e.what() << "\n";
    }

    // List records after registration
    auto updated = client.dns().list();
    std::cout << "\nDNS records after registration: " << updated.count << "\n";
    for (const auto& record : updated.records) {
        std::cout << "  " << record.name << " -> " << record.address << "\n";
    }

    return 0;
}
